src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/persist/SQLParsingUtil.java

```diff
@@ -37,14 +37,15 @@
    * @throws IllegalArgumentException throws an exception if we can't find the name in the field
    */
   private static Record getRecordByName(Result<Record> result, Field<String> field, String name) throws IllegalArgumentException {
```

Comment:
If we don't expect `result` to be `null`, we should add the notnull annotation for that argument.

Reply:
Disagree, we explicitly included logic dealing with null results which results in a logging statement and a particular double.NaN return statement from the public caller. Adding the annotation to this private method would only shift the (result == null) logic to the public function and slightly increase the overall line count.

Reply:
Throwing an exception is a runtime validation but I think we should try and prefer static checks over them as this way the static analysis tools can do the work for you before the code goes to prod. Anyways, the @nonnull is not enforced by the compiler. So, we can ignore it.